server/PipelineRunner.js

```diff
@@ -300,9 +300,11 @@ function getTimeNow() {
 function createJob({key, sampleName}) {
     let job = {};
     /* basic information */
-    job.sample_name = sampleName;
-    job.barcodes = global.datastore.getBarcodesForSampleName(sampleName);
-    job.samples = `{${job.sample_name}: [${job.barcodes}]}`;
+    if(sampleName){ //run-per-sample
```

Comment:
why is this if needed? what will happen otherwise?

Reply:
it is really not needed, I 'll remove it.

Reply:
you will get this error otherwise: ![why](https://user-images.githubusercontent.com/42906271/116856184-fd9c8880-abfa-11eb-856d-b35aedf2e7c4.png)

Reply:
But does this happen only in our modification of the code or in general? What are the exact circumstances the function is called with empty sampleName?

Reply:
the sampleName is relevant only when we want to create a job for run-per-sample pipeline. In original rampart version, only the annotation pipeline was not a run-per-sample one. we created another pipeline, which is not run-per-sample, but we want to be able to trigger it manually from UI, not automatically when the server starts. Therefore we added this if - the createJob method is used in listeners defined in `socket.js`, where we are adding a job to queue. Run-per-sample pipelines will not be affected by this change. When we would like to add another pipeline which should be run on all samples, we can reuse this modified createJob method. When the pipeline is to be run for all the samples, we dont have any sampleName defined, so that the call of `global.datastore.getBarcodesForSampleName(sampleName)` method will fail, and we will get the fatal error mentioned above - and calling this method is irrelevant for pipeline which is not run-per sample...

Reply:
Ok, so perhaps we can clarify, before the if statement
`// only pipelines which are run per sample have sampleName set`